### PR TITLE
CT 116: Add VET TEC logo to GI Bill Comparison Tool

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a Comparison Tool user, I need to see the VET TEC logo in GI Bill Comparison Tool.

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19183

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19312

Includes revert of https://github.com/department-of-veterans-affairs/vets-website/pull/10461

## Testing done
Local manual, unit, and e2e tests pass

## Screenshots

Screen Shot 2019-07-23 at 11 02 20 AM
Screen Shot 2019-07-23 at 11 02 14 AM
Screen Shot 2019-07-23 at 11 02 05 AM
Screen Shot 2019-07-23 at 11 01 57 AM
Screen Shot 2019-08-15 at 12 32 40 PM
Screen Shot 2019-08-15 at 12 32 47 PM

## Acceptance criteria
- [x] The VET TEC Logo is not displayed when the answer to the "Type of institution" was "Schools".
- [x] The VET TEC Logo is not displayed when the answer to the "Type of institution" was "Employers (On-the-job training [OJT], Apprenticeships)".
- [x] The VET TEC logo is displayed in the top right corner of the page when the user selects "VET TEC training providers" as the answer to "Type of institution".
- [x] The VET TEC logo is 85px high.
- [ ] The VET TEC logo is not displayed in production
- [ ] The VET TEC logo is displayed on profile page

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Similar to your other PR, can you use the shortened paths for your imports. Please move your inline style to a css class and refer to the class.

File: src/applications/gi/containers/SearchPage.jsx [hunk 1]
```diff
@@ -230,14 +231,53 @@ export class SearchPage extends React.Component {
     );
 
     const searchResults = this.searchResults();
+
     return (
       <ScrollElement name="searchPage" className="search-page">
+        {!environment.isProduction() &&
```
Comment (zurbergram) [inline src/applications/gi/containers/SearchPage.jsx]: why is this here twice?
if it needs to be twice extract out into a renderLogo function

**Review by zurbergram — DISMISSED** [review]
LGTM

**Review by shawkey-daniel — DISMISSED** [review]
LGTM

**Review by sonntag-adam — DISMISSED** [review]

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/frontend-review-group Could I get a code review on this at your earliest convenience. Thank you!

File: src/applications/gi/components/search/TypeOfInstitutionFilter.jsx [hunk 2]
```diff
@@ -58,13 +47,35 @@ class TypeOfInstitutionFilter extends React.Component {
...
+              <img
+                className="vettec-logo vads-u-padding-top--0p5 vads-u-margin-bottom--neg1"
+                src="/img/logo/vet-tec-logo.png"
+                alt="Vet Tec Logo"
+              />
+            </span>
+            {/* mobile size */}
+            <span className="vads-u-margin-bottom--1  vads-u-display--block small-screen:vads-u-display--none">
+              <img
+                className="vettec-mobile-logo"
```
Comment (cvalarida) [inline src/applications/gi/components/search/TypeOfInstitutionFilter.jsx]: It looks like we've got a few things going on here:
1. We're rendering two `<span>` DOM nodes
2. The  `<span>`s have switching logic to determine which one is shown based on the viewport size
3. The `<span>`s also have differing margins
4. Inside the `<span>`s, we've got ``s that are the same except the class names
5. These class names differ by controlling some margin and the size of the image

Does it maybe make more sense to do this all in SASS using screen size mixins and custom classes?

We'd then end up with something more like:
```jsx
{isVetTecSelected(this.props) &&
  <span className="vettec-logo-container">
    <img
      className="vettec-logo"
      src="/img/logo/vet-tec-logo.png"
      alt="Vet Tec Logo"
    />
  />
}
```

I'm not super keen on using JavaScript to solve a CSS problem, y'know?

File: src/applications/gi/containers/SearchPage.jsx [hunk 3]
```diff
@@ -226,68 +226,102 @@ export class SearchPage extends React.Component {
...
+              <img
+                className="vettec-mobile-logo"
+                src={require('site/assets/img/logo/vet-tec-logo.png')}
+                alt="Vet Tec Logo"
+              />
+            </div>
+            <div className="row">
+              <div className="column">
+                <div className="vads-l-row vads-u-justify-content--space-between vads-u-align-items--flex-end">
+                  <div className="vads-l-col--10 site-grid-example">
```
Comment (cvalarida) [inline src/applications/gi/containers/SearchPage.jsx]: Where does this `site-grid-example` come from?

File: src/applications/gi/containers/SearchPage.jsx [hunk 4]
```diff
@@ -226,68 +226,102 @@ export class SearchPage extends React.Component {
...
-          />
+          <div>
+            <div className="vads-u-display--block small-screen:vads-u-display--none vettec-mobile-logo-container">
+              <img
+                className="vettec-mobile-logo"
+                src={require('site/assets/img/logo/vet-tec-logo.png')}
+                alt="Vet Tec Logo"
+              />
+            </div>
+            <div className="row">
```
Comment (cvalarida) [inline src/applications/gi/containers/SearchPage.jsx]: Is there a lighter way to do this without the use of a grid? I think this is semantic, but feels heavy for what it's doing, y'know?

File: src/applications/gi/containers/SearchPage.jsx [hunk 5]
```diff
@@ -226,68 +226,102 @@ export class SearchPage extends React.Component {
...
-            }
-            filters={filters}
-            toggleFilter={this.props.toggleFilter}
-            searchResults={searchResults}
-            eligibility={this.props.eligibility}
-            showModal={this.props.showModal}
-            eligibilityChange={this.props.eligibilityChange}
-          />
+          <div>
+            <div className="row">
```
Comment (cvalarida) [inline src/applications/gi/containers/SearchPage.jsx]: Same question here: Does it make sense to use a grid for this?

File: src/applications/gi/sass/partials/_gi-search-page.scss [hunk 6]
```diff
@@ -36,7 +36,8 @@
     }
   }
 
-  .filter-button, .results-button {
+  .filter-button,
+  .results-button {
```
Comment (cvalarida) [inline src/applications/gi/sass/partials/_gi-search-page.scss]: I appreciate the cleanup in this file. :+1:

File: src/applications/gi/sass/partials/_gi-search-page.scss [hunk 7]
```diff
@@ -159,6 +161,25 @@
         margin-right: 0.4em;
       }
     }
+  }
+
+  .vettec-logo-container {
+    margin-top: -32px;
+    margin-bottom: 16px;
```
Comment (cvalarida) [inline src/applications/gi/sass/partials/_gi-search-page.scss]: For consistency, can we express the margins and paddings in here in `em` instead of `px`?

**Review by cvalarida — COMMENTED** [review]
I had a few concerns, mostly dealing with the use of SASS.

**Review by rianfowler — DISMISSED** [review]
Pending review by DSVA

File: src/applications/gi/components/search/TypeOfInstitutionFilter.jsx [hunk 8]
```diff
@@ -58,13 +47,35 @@ class TypeOfInstitutionFilter extends React.Component {
...
+          label="Type of institution"
+          name="category"
+          options={options}
+          value={this.props.category}
+          onChange={this.props.onChange}
+        />
+        {isVetTecSelected(this.props) && (
+          <div>
+            {/* full size */}
+            <span className="vads-u-margin-x--neg1  vads-u-display--none small-screen:vads-u-display--block">
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/components/search/TypeOfInstitutionFilter.jsx]: yes

File: src/applications/gi/sass/partials/_gi-search-page.scss (hunk 7 above)
Comment (zurbergram) [reply src/applications/gi/sass/partials/_gi-search-page.scss]: Tried to switch to em and this caused the image to be skewed incorrectly

File: src/applications/gi/components/search/TypeOfInstitutionFilter.jsx (hunk 2 above)
Comment (zurbergram) [reply src/applications/gi/components/search/TypeOfInstitutionFilter.jsx]: used sass logic to change height and width on landing page

File: src/applications/gi/containers/SearchPage.jsx (hunk 3 above)
Comment (zurbergram) [reply src/applications/gi/containers/SearchPage.jsx]: bad copy paste

File: src/applications/gi/sass/partials/_gi-search-page.scss (hunk 7 above)
Comment (zurbergram) [reply src/applications/gi/sass/partials/_gi-search-page.scss]: done

File: src/applications/gi/containers/SearchPage.jsx (hunk 4 above)
Comment (zurbergram) [reply src/applications/gi/containers/SearchPage.jsx]: old code, removing didn't affect display

File: src/applications/gi/containers/SearchPage.jsx (hunk 5 above)
Comment (zurbergram) [reply src/applications/gi/containers/SearchPage.jsx]: old code, removing didn't affect display

File: src/applications/gi/sass/partials/_gi-vet-tec.scss [hunk 9]
```diff
@@ -0,0 +1,25 @@
+.gi-app {
+  .vettec-logo {
+    height: 70px;	
+    width: 146px;	
+  }
+
+  // this will only affect wide screens
+  @media screen and (min-width: 481px) {
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/gi/sass/partials/_gi-vet-tec.scss]: done

**Review by sonntag-adam — APPROVED** [review]

**Review by delli-gatti-michael — APPROVED** [review]
Approved by QA

**Review by devin-mccurdy — APPROVED** [review]
LGTM

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Comment by batemapf** [comment]
DSVA is allowing the use of the VET TECH logo on VA.gov at the request of OPIA. This is an exception to the [VA.gov Design System's provisions on logos](https://design.va.gov/design/logos), conflicts with [VA Graphics Standards'](https://www.va.gov/vapubs/viewPublication.asp?Pub_ID=683&FType=2) guidance on the VA brand, and is not aligned with the brand consolidation effort that is critical to the success of the VA Digital Modernization Strategy.

@department-of-veterans-affairs/frontend-review-group please conduct a review on this PR per our [guidelines](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/Practice%20Areas/Engineering/code_review_guidelines.md).

cc @davidsbao @chrisj-usds @leahbannon

**Review by rianfowler — APPROVED** [review]
